File: py/myclient.py
from __future__ import print_function

import grpc

import mybooks_pb2
import mybooks_pb2_grpc

'''
python3 -m pip install --upgrade pip
python3 -m pip install grpcio
python3 -m pip install grpcio-tools
python3 -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./mybooks.proto 
'''

channel = grpc.insecure_channel('localhost:50051')
stub = mybooks_pb2_grpc.BookServiceStub(channel)


def listBooks():
    global stub
    result = stub.List( mybooks_pb2.BookIdRequest() )
    print( result )

def getBookId(xid):
    global stub
    try:
        result = stub.Get( mybooks_pb2.Book( id=xid ))
        print( result )
    except Exception as err :
        print( err )
    
def postBook( xid, xtitle, xauthor ):
    global stub
    try:
        result = stub.Post( mybooks_pb2.Book(id=xid, title=xtitle , author=xauthor) )
        #print( result )
    except Exception as err:
        print(err)


def delBook( xid ):
    global stub
    try:
        result = stub.Delete( mybooks_pb2.BookIdRequest( id=xid))
    except Exception as err:
        print( err)
        

if __name__ == '__main__':
    #listBooks()
    #getBookId( 123 )
    postBook( 222, "test02" , "alex")
    #delBook( 123 )
    #getBookId(123)
    listBooks()

File: myclient.js

var grpc = require('grpc');
var booksProto = grpc.load('mybooks.proto');
var client = new booksProto.books.BookService('0.0.0.0:50051', grpc.credentials.createInsecure());

function listBooks() {
    client.list({}, function(error, result) {
        console.log("listBook-->");
        if (error){
            console.log( 'Error:', error);
        }else{            
            console.log( result) ;
        }
    });
}

function getBookId( obj ){

    client.get(  {id: parseInt( obj.id) } , function( error , result){
        console.log("getBookId-->");
        if ( error ){
            console.log( 'Error:' , error );
        }else{
            console.log( result )
        }
   });
}

function postBook(obj){
    var book = {  
        id: obj.id,
        title: obj.title,
        author: obj.author
    };
    
    client.post( book , function( error , result){
        console.log("postBook-->");
        if (error){
            console.log( 'Error:' , error);
        }else{
            console.log( result );
        }
    });
}

function putBook( obj ){
    var book = {  
        id: obj.id,
        title: obj.title,
        author: obj.author
    };
    
    client.put( book , function(error , result){
        console.log("putBook-->");
        if (error){
            console.log( 'Error:' , error);
        }else{
            console.log( result );
        }
    });
}

function delBook( obj ){
    client.delete( { id: obj.id } , function(error , result){
        console.log("delBook-->");
        if (error){
            console.log( 'Error:' , error);
        }else{
            console.log( result );
        }
    });
}

//listBooks
//listBooks();

//Get 
//getBookId( {id:123} );

//Post
postBook( { id:222 , title:'hello' , author:"alex"});

//Put
//putBook( { id: 123 , title:'hello' , author:"alex"});

//Delete
//delBook( {id:123} );

listBooks();
